frontend/sozisel-app/src/components/Register/Register.tsx

```diff
@@ -0,0 +1,130 @@
+import "./register.scss";
+
+import * as Yup from "yup";
+
+import { ReactElement, useState } from "react";
+
+import Button from "../utils/Button/Button";
+import Card from "../utils/Card/Card";
+import ErrorMessage from "../utils/Input/ErrorMessage";
+import Input from "../utils/Input/Input";
+import Navbar from "../Navbar/Navbar";
+import conference_img from "../../assets/conference_img.png";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router";
+import { useRegisterMutation } from "../../graphql";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+interface RegisterFormData {
+  email: string;
+  firstName: string;
+  lastName: string;
+  password: string;
+  confirmPassword: string;
+}
+
+const fieldRequiredError = "To pole jest wymagane!";
+const passwordsNotMatchError = "Podane hasła nie są identyczne!";
+
+const registerSchema = Yup.object().shape({
+  email: Yup.string()
+    .email("Niepoprawny format email!")
+    .required(fieldRequiredError),
+  firstName: Yup.string().required(fieldRequiredError),
+  lastName: Yup.string().required(fieldRequiredError),
+  password: Yup.string().required(fieldRequiredError),
+  confirmPassword: Yup.string().required(fieldRequiredError),
+});
+
+export default function Register(): ReactElement {
+  const [registerMutation] = useRegisterMutation();
+  const [error, setError] = useState<string>("");
+  const history = useHistory();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: yupResolver(registerSchema),
+  });
+
+  const onSubmit = (registerFormData: RegisterFormData) => {
+    if (registerFormData.password !== registerFormData.confirmPassword) {
+      setError(passwordsNotMatchError);
+    } else {
+      registerMutation({
+        variables: {
```

Comment:
```suggestion
        variables: { ...registerFormData }
```

Reply:
I cannot do that, because `registerFormData` contains one additional property `confirmPassword`, and a react hook mutation doesn't recognize it and throws an error.